### ISSUE TITLE
compile docs from jsdoc

### ISSUE THREAD

**Issue by samuelstroschein** [opral/lix-sdk#89]
### Context

We can compile a reference documentation via [https://typedoc.org/](https://typedoc.org/). 

### Proposal

Implement [https://typedoc.org/](https://typedoc.org/) for the lix SDK. 

* only have to write reference documentation once 
* nudges us to write good jsdoc

**Comment by samuelstroschein** [comment]
@felix.haeberle is this task something you are interested in ?> 

* contributing to the docs as starting point to go deeper into lix

**Comment by samuelstroschein** [comment]
@felixhaeberle do this now? I just found myself looking for an api reference for the key_value table and couldn't find it. 

API reference from the code makes so much sense. No dual maintenance of api docs + real code.<br><br>I implemented this in Paraglide JS [https://typedoc-plugin-markdown.org/](https://typedoc-plugin-markdown.org/) Works like a charme.

**Tasks**

1. Setup [https://typedoc-plugin-markdown.org/](https://typedoc-plugin-markdown.org/)
2. Emit into a `api-reference` folder 
3. Sync the api-reference folder to [https://github.com/opral/lix-sdk/tree/main/api-reference](https://github.com/opral/lix-sdk/tree/main/api-reference) by creating a new sync workflow similar to  [https://github.com/opral/lix-sdk/blob/main/.github/workflows/readme-sync.yml](https://github.com/opral/lix-sdk/blob/main/.github/workflows/readme-sync.yml)

**Comment by felixhaeberle** [comment]
Yes. Lets do this, I will focus on this later in the day, after clearing some minor Sherlock bugs.

**Comment by samuelstroschein** [comment]
implemented

**Comment by felixhaeberle** [comment]
We have the individual files now also!:)

![Image](https://github.com/user-attachments/assets/cd2cc1a3-6a63-4fd0-b60d-e79b0d1c29b2)
